src/ui/public/query_bar/directive/query_bar.js

```diff
@@ -15,55 +20,107 @@ module.directive('queryBar', function () {
       query: '=',
       appName: '=?',
       onSubmit: '&',
-      disableAutoFocus: '='
+      disableAutoFocus: '=',
+      indexPatterns: '='
     },
     controllerAs: 'queryBar',
     bindToController: true,
-    controller: callAfterBindingsWorkaround(function ($scope, config, PersistedLog) {
+
+    controller: callAfterBindingsWorkaround(function ($scope, $element, $http, $timeout, config, PersistedLog, indexPatterns) {
       this.appName = this.appName || 'global';
       this.availableQueryLanguages = queryLanguages;
       this.showLanguageSwitcher = config.get('search:queryLanguage:switcher:enable');
 
+      let persistedLog;
+
+      this.getIndexPatterns = () => {
+        if (compact(this.indexPatterns).length) return Promise.resolve(this.indexPatterns);
+        return Promise.all([indexPatterns.getDefault()]);
```

Comment:
Is it necessary to grab the default here? Won't each individual app do that if it's the right thing to do? I was thinking if no indexPattern was passed in, we just wouldn't provide suggestions. If we didn't have to grab the default, I think it would remove the need to deal with promises here?